src/act.drive.cpp

```diff
@@ -1268,19 +1268,23 @@ ACMD(do_driveby)
     send_to_char("How do you expect to driveby from a stationary vehicle?\r\n", ch);
     return;
   }
-
+ 
   if ((wielded = GET_EQ(ch, WEAR_WIELD))) {
-    if (!(GET_OBJ_VAL(wielded, 4) == SKILL_PISTOLS || GET_OBJ_VAL(wielded, 4) == SKILL_SMG)) {
-      send_to_char(ch, "You can only do a driveby with a pistol or SMG.\r\n");
+    if (GET_OBJ_TYPE(wielded) != ITEM_WEAPON) {
+      send_to_char(ch, "You have to be wielding a weapon!"
+      return;
+    }
+    if ((IS_OBJ_STAT(wielded, ITEM_EXTRA_TWOHANDS) || (!WEAPON_IS_GUN(wielded) && ch->in_veh->type = !VEH_BIKE)))  {
+      send_to_char(ch, "You can only do a driveby with a one handed weapon in a vehicle and a melee weapon while on a bike!\r\n");
```

Comment:
This message is confusing, try splitting it with an if-check and giving them the proper message for the vehicle they're in.